### PR TITLE
Darek

### PR DESCRIPTION
**Pull request by chefdarek**
Cleaning V.2

### CODE REVIEW

**Comment by axrd** [comment]
# Module 01 - Making Datasets, Avoiding Leakage

*Overall:* You're off to a solid start with the Lending Club data! I can tell you've gotten really good at wrangling data into the shape and format that you want it 👍 
*Feedback*: Get in the habit of being explicit about what you're doing and why. If you can explain your reasoning to a non-technical reader, you're quality of communication accounts for virtually every stakeholder you'll engage with. Leverage comments and text cells more.

## Get the data 
- Glob FTW! For future reference, it can be easier to assign your path to a variable and then use that to work with multiple files inside the same notebook. For example:  
`path =  '/Users/YourUserName/Documents/etc/etc/FILES!'`

## Choose number of observations
- I see that you split the data but it would be great to add a text cell or some commenting to explain how/why you decided to do so (code etiquette). 
- Nice job keeping it down to 141MB

## Save your Data for reuse
- 👍 

## Choose Features
- Great job killing whitespace and cleaning emp_length. Nicely done using anonymous functions
- Same as above. Document your choices more explicitly with text cells / comments.

**Comment by axrd** [comment]
# SC Review 

*Overall:* You've come a helluva long way in 9 weeks. 
*Feedback*: N/A | Keep doing what you're doing.  

## Making Datasets, Avoiding Leakage
- Glob is a superpower. Remember that tomorrow in case you need to combine multiple csvs.  
- Key point of Data Hygiene in predictive modeling: leakage can easily lead you astray so proactively prevent it. Removing features is one really effective way to go about it. Consider Time-Series!

## Start with Baselines
- Not an explicit learning objective of this Sprint but so important that it's worth reiterating 👍 

## Class Imbalances
- When confronting classification problems, accuracy may not be a good measure because it doesn't factor True Positive rate. But the threshold for this is _contextual_. You want to bear in mind the *consequences* of your choices when factoring this in. Par for the course re: Predictive Modeling & Decision Making.
- If you determine that you can't afford false positives, use ROC AUC as a more appropriate metric for evaluating your model. By plotting the true positive rate vs the false positive rate at different thresholds, you can better understand your model's predictive power. 

## Making & Explaining Decisions
- The whole point of Data Science (if you can reduce it down to just one) is to draw actionable insights. That means making decisions. In order to better calibrate that decision-making, we leverage Expected Values, and Shapley Value Plots. Review the lecture notebook for links to dive deeper into both of these topics if you wish!

## Gradient Descent
- How can you programaticaly minimize a value you care about? Gradient Descent refers to a method that iteratively discovers minima. The trick is understanding if they are local or global, for one. For another, computation is not free. The time and effort it takes your model to minimize it's errors is a consideration you must make. Calibrate the learning_rate as one knob/dial.
